sympy/functions/elementary/hyperbolic.py

```diff
@@ -336,12 +346,25 @@ def _eval_as_leading_term(self, x):
             return self.func(arg)
 
     def _eval_is_real(self):
-        return self.args[0].is_real
+        from sympy import pi
+        arg = self.args[0]
+
+        # `cosh(x)` is real for real and purely imaginary `x`
```

Comment:
Oh, that was the 'and' from a literature perspective. :laughing: 
Will change it.